File: app/api/schemas/equipment.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional, List
from app.models.equipment import EquipmentStatus  # Импортируем из модели

class EquipmentCreate(BaseModel):
    name: str
    user_id: int
    category_id: int
    description: Optional[str] = None
    quantity: int = 1
    latitude: Optional[float] = None
    longitude: Optional[float] = None

class EquipmentResponse(BaseModel):
    id: int
    name: str
    user_id: int
    category_id: int
    category_name: str
    status: EquipmentStatus
    description: Optional[str]
    quantity: int
    created_at: datetime
    latitude: Optional[float]
    longitude: Optional[float]
    rejection_reason: Optional[str] = None
    moderated_at: Optional[datetime] = None
    moderated_by: Optional[int] = None
    
    model_config = ConfigDict(from_attributes=True)

class EquipmentUpdate(BaseModel):
    name: Optional[str] = None
    category_id: Optional[int] = None
    description: Optional[str] = None
    quantity: Optional[int] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    status: Optional[EquipmentStatus] = None

class EquipmentStatusUpdate(BaseModel):
    status: EquipmentStatus

class EquipmentRejectUpdate(BaseModel):
    reason: Optional[str] = None

class EquipmentQuantityUpdate(BaseModel):
    quantity: int

class EquipmentListResponse(BaseModel):
    equipments: List[EquipmentResponse]
    total: int
    skip: int
    limit: int

class EquipmentDetailedResponse(EquipmentResponse):
    display_status: str
    
    model_config = ConfigDict(from_attributes=True)

class EquipmentSearchParams(BaseModel):
    category_id: Optional[int] = None
    user_id: Optional[int] = None
    status: Optional[EquipmentStatus] = None
    name: Optional[str] = None
    skip: Optional[int] = 0
    limit: Optional[int] = 100

class EquipmentStatsResponse(BaseModel):
    total: int
    moderation: int
    approved: int
    rejected: int

File: nko-admin-panel/src/app/services/equipment.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface Equipment {
  id?: number;
  name: string;
  user_id: number;
  category_id: number;
  category_name: string;
  status: string; // 'moderation' | 'approved' | 'rejected'
  description?: string;
  quantity: number;
  created_at?: string;
  latitude?: number;
  longitude?: number;
  display_status?: string;
  rejection_reason?: string;
  moderated_at?: string;
  moderated_by?: number;
}

export interface EquipmentListResponse {
  equipments: Equipment[];
  total: number;
  skip: number;
  limit: number;
}

export interface EquipmentUpdate {
  name?: string;
  category_id?: number;
  description?: string;
  quantity?: number;
  latitude?: number;
  longitude?: number;
  status?: string;
}

export interface EquipmentSearchParams {
  category_id?: number;
  user_id?: number;
  status?: string;
  name?: string;
}

export interface EquipmentStats {
  total: number;
  moderation: number;
  approved: number;
  rejected: number;
}

@Injectable({
  providedIn: 'root'
})
export class EquipmentService {
  private baseUrl = environment.apiUrl + environment.endpoints.equipment;

  constructor(private http: HttpClient) {}

  getEquipment(
    skip: number = 0,
    limit: number = 100,
    status?: string
  ): Observable<EquipmentListResponse> {

    let params = new HttpParams()
      .set('skip', skip.toString())
      .set('limit', limit.toString());

    if (status) {
      params = params.set('status', status);
    }

    return this.http.get<EquipmentListResponse>(this.baseUrl, { params });
  }

  approveEquipment(equipmentId: number): Observable<Equipment> {
    return this.http.patch<Equipment>(`${this.baseUrl}/${equipmentId}/approve`, {});
  }

  rejectEquipment(equipmentId: number, reason: string = ''): Observable<Equipment> {
    return this.http.patch<Equipment>(`${this.baseUrl}/${equipmentId}/reject`, { reason });
  }
}